basyx.components/basyx.components.docker/basyx.components.AASServer/src/main/java/org/eclipse/basyx/components/aas/mongodb/MongoDBAASAggregator.java

```diff
@@ -417,31 +419,18 @@ public void setRegistry(IAASRegistry registry) {
 	 * @deprecated This method is used with the old, deprecated Constructors. Use
 	 *             {@link MongoDBAASServerComponentFactory} instead
 	 */
-	@SuppressWarnings("deprecation")
 	@Deprecated
 	public void setConfiguration(BaSyxMongoDBConfiguration config) {
-		// set mongoDB configuration
-		this.config = config;
 		MongoClient client = MongoClients.create(config.getConnectionUrl());
-		this.mongoOps = new MongoTemplate(client, config.getDatabase());
-		this.aasCollection = config.getAASCollection();
-		this.smCollection = config.getSubmodelCollection();
+		setMongoDBConfiguration(config, client);
 
-		// Create API factories with the given configuration
-		this.aasApiProvider = aas -> {
-			MongoDBAASAPI api = new MongoDBAASAPI(config, aas.getIdentification().getId());
-			api.setAAS(aas);
-			return api;
-		};
-		this.smApiProvider = sm -> {
-			MongoDBSubmodelAPI api = new MongoDBSubmodelAPI(config, sm.getIdentification().getId());
-			api.setSubmodel(sm);
-			return api;
-		};
+		this.aasApiProvider = new MongoDBAASAPIFactory(config, client);
+		this.smApiProvider = new MongoDBSubmodelAPIFactory(config, client);
 	}
 
 	private void setMongoDBConfiguration(BaSyxMongoDBConfiguration config, MongoClient client) {
```

Comment:
Would it be better to have a method "setMongoClient(config)" which also creates the client?

Reply:
This is already covered by the setConfiguration method. It creates the client if it was not already passed via the constructor.

Reply:
Ok.